lib/Teng/Plugin/SearchJoined/Iterator.pm

```diff
@@ -43,9 +43,14 @@ sub _seperate_rows {
     my $name_sep = quotemeta $self->{teng}{sql_builder}{name_sep};
     my $i = 0;
     for my $field (@{ $self->{fields} }) {
-        my $value = $row->[$i++];
+        my $value = $row->[$i];
         my ($table, $column) = split /$name_sep/, $field;
+        if (!$column) {
+            $column = $self->{sth}->{NAME_lc}->[$i];
+            $table  = $self->{table_names}->[0]; # base_table
```

Comment:
Surely it is I think a good idea. .
But, for the following reasons, 
I wanted to remains `base_table`.

* schema definition is needed when passing `new_row_from_hash` 
  * https://github.com/nekokak/p5-Teng/blob/master/lib/Teng.pm#L646-L647
* I want to similar shape and when you use the `+columns` in Teng

e.g.  Teng +columns example
```perl

use Test::More;
use DBI;
use Teng;
use Teng::Schema::Declare;

my $dbh = DBI->connect('dbi:mysql:test', 'root', '');

my $schema = schema {
    table {
        name "foo";
        columns qw( id );
    };
} "MyDB::Schema";

my $teng = Teng->new(
    dbh    => $dbh,
    schema => $schema,
);

my $row = $teng->single(foo => +{}, { '+columns' => [\'1+1 as hello']});
is $row->hello, 2;

done_testing;
```